sdk/core/azure_core/src/task/standard_spawn.rs

```diff
@@ -84,36 +96,58 @@ impl TaskSpawner for StdSpawner {
         {
             let join_state = Arc::new(Mutex::new(ThreadJoinState::default()));
             {
-                let join_state_clone = join_state.clone();
-                let js = join_state
-                    .lock()
-                    .map_err(|e| format!("Failed to lock join state: {:?}", e));
-                let Ok(mut js) = js else {
-                    return Box::pin(async { Err("Failed to lock join state".into()) });
-                };
-                js.join_handle = Some(thread::spawn(move || {
-                    // Create a local executor
-                    let mut local_pool = LocalPool::new();
-                    let spawner = local_pool.spawner();
+                let js = join_state.lock();
+                match js {
+                    Ok(mut js) => {
+                        let join_state_clone = join_state.clone();
+                        js.join_handle = Some(thread::spawn(move || {
+                            // Create a local executor
+                            let mut local_pool = LocalPool::new();
+                            let spawner = local_pool.spawner();
 
-                    // Spawn the future on the local executor
-                    let future_handle = spawner
-                        .spawn_with_handle(f)
-                        .expect("Failed to spawn future");
+                            // Spawn the future on the local executor
```

Comment:
Nit: the deep nesting here is a bit hard to follow. I wouldn't hold up the PR for this, but maybe consider refactoring using the `let..else` syntax to unnest e.g.,

```rust
let Ok(future_handle) = spawn_result else {
  return Err(crate::Error::mesage(...));
};

// ...
```

Reply:
The only negative on that is that it loses the ability to report the actual error from the spawn call. But that may not be significant.

Reply:
Cleaned up a few of these.

Reply:
For posterity, as we discuss in person, that's probably fine because your error strings are unique to the point of tracking down *where* the error occurs and, as you pointed out, in most cases it's a failure to get a lock...which should be rare anyway.

If we ever revisit it, you can do something like a match with an explicit return out of the function or even a `map_err` in many cases.